### PR TITLE
feat: Consent page design and external client header

### LINKED ISSUE #389: [Task] Implement the Consent page Design
[See Figma Auth design](https://www.figma.com/file/1TeGq312pT4ZIiRn5u6iYU/03_ipsoon%26lesdevs?type=design&node-id=1093-14625&t=90OCyXTAXO3umyjY-0)

### PR DESCRIPTION
**Pull request by taorepoara**
Closes #389 

## Description of the changes
Implements the consent page design and the external app header.

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [x] I made my own code-review before requesting one

### I included unit tests that cover my changes
  - [ ] 👍 yes
  - [x] 🙅 no, because they aren't needed
  - [ ] 🙋 no, because I need help

### I added/updated the documentation about my changes

- [ ] 📜 README.md
- [x] 📕 docs/*.md
- [ ] 📓 docs.lenra.io
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by jonas-martinez** [comment]
It seems that the app name is not shown on the consent page. Is it a problem on my side or on your side ?

image

**Review by jonas-martinez — CHANGES_REQUESTED** [review]
See my comment above

**Comment by taorepoara** [comment]
@jonas-martinez How did you create the OAuth client ?

Did you use the dedicated Elixir task ?

**Comment by jonas-martinez** [comment]
No, it was why it did not show the app name on the screen. It works now :smiley:

**Review by jonas-martinez — APPROVED** [review]

File: apps/identity_web/lib/identity_web/templates/user_consent/consent.html.heex [hunk 1]
```diff
@@ -1,35 +1,29 @@
...
+  <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/assets/form.css")} media="print" onload='this.media="all"'/>
+  <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/css/consent.css")} media="print" onload='this.media="all"'/>
+  <script defer phx-track-static type="text/javascript" src={Routes.static_path(@conn, "/assets/form.js")}></script>
+  <%= oauth_header "consent", @client %>
   <.form let={f} for={@conn} action={Routes.user_consent_path(@conn, @action)} >
-      <%= hidden_input(f, :accept, value: true) %>
-      <%= if assigns[:consent_challenge] do %>
-        <%= hidden_input f, :consent_challenge, value: @consent_challenge %>
+    <p><%= raw(gettext("Hi <strong>%{email}</strong> !", email: @user.email)) %></p>
+    <p><%= raw(gettext("<strong>%{app_name}</strong> is requesting access to your Lenra’s account.", app_name: get_client_name(@client))) %></p>
```
Comment (jonas-martinez) [inline apps/identity_web/lib/identity_web/templates/user_consent/consent.html.heex:8]:
```suggestion
    <p><%= raw(gettext("<strong>%{app_name}</strong> is requesting access to your Lenra account.", app_name: get_client_name(@client))) %></p>
```

**Review by jonas-martinez — CHANGES_REQUESTED** [review]

**Review by jonas-martinez — APPROVED** [review]

**Pull request merged**